types/setting.go

```diff
@@ -1781,10 +1781,11 @@ var (
 type NodeDownPodDeletionPolicy string
 
 const (
-	NodeDownPodDeletionPolicyDoNothing                             = NodeDownPodDeletionPolicy("do-nothing") // Kubernetes default behavior
-	NodeDownPodDeletionPolicyDeleteStatefulSetPod                  = NodeDownPodDeletionPolicy("delete-statefulset-pod")
-	NodeDownPodDeletionPolicyDeleteDeploymentPod                   = NodeDownPodDeletionPolicy("delete-deployment-pod")
-	NodeDownPodDeletionPolicyDeleteBothStatefulsetAndDeploymentPod = NodeDownPodDeletionPolicy("delete-both-statefulset-and-deployment-pod")
+	NodeDownPodDeletionPolicyDoNothing                                             = NodeDownPodDeletionPolicy("do-nothing") // Kubernetes default behavior
+	NodeDownPodDeletionPolicyDeleteStatefulSetPod                                  = NodeDownPodDeletionPolicy("delete-statefulset-pod")
+	NodeDownPodDeletionPolicyDeleteDeploymentPod                                   = NodeDownPodDeletionPolicy("delete-deployment-pod")
+	NodeDownPodDeletionPolicyDeleteBothStatefulsetAndDeploymentPod                 = NodeDownPodDeletionPolicy("delete-both-statefulset-and-deployment-pod")
+	NodeDownPodDeletionPolicyDeleteBothStatefulsetAndDeploymentPodAndAnnotatedPods = NodeDownPodDeletionPolicy("delete-both-statefulset-and-deployment-pod-and-annotated-pods")
```

Comment:
Should we retain one? Given that the new configuration is essentially a superset of the current one, `delete-both-statefulset-and-deployment-pod`.

Additionally, we might consider implementing this resilience for all pods managed by controllers, rather than exclusively for SS and DS workloads, as we currently do for the setting `auto-delete-pod-when-volume-detached-unexpectedly` handling mentioned below. @derekbit @longhorn/maintainer  WDYT?

[/controller/kubernetes_pod_controller.go#L634-L636](https://github.com/longhorn/longhorn-manager/blob/e6647b81bf21f5337dcb710a65e25ad7b941ef0a/controller/kubernetes_pod_controller.go#L634-L636)
```go
	if ownerRef := metav1.GetControllerOf(pod); ownerRef == nil || kc.isControllerInBlacklist(ownerRef) {
		return nil
	}
```